pkg/util/filesystem/permission_nowindows.go

```diff
@@ -0,0 +1,22 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+// +build !windows
+
+package filesystem
+
+import "os"
+
+// Handle permissions for Unix and Windows
+type Permission struct{}
+
+// Create a new instance of `Permission`
+func NewPermission() (*Permission, error) {
+	return &Permission{}, nil
+}
+
+// Set the permission of `path` to the current user and current group.
+func (p *Permission) SetPermToCurrentUserAndGroup(path string) error {
+	return os.Chmod(path, 0700)
+}
```

Comment:
Comparing the Windows and linux implementation, why aren't we doing the same and changing the user/group ?

This doesn't use the current user/group. If the file is own by another user/group we would just change the right not the group or user.

Reply:
>why aren't we doing the same and changing the user/group ?
The purpose of this method is to restrict the access of the folder to the Agent or an administrator only.